apache-pulsar-transport-binding.md

```diff
@@ -0,0 +1,228 @@
+value is made available as [UTF-8][RFC3629] encoded JSON text.
+
+## 3. Apache Pulsar Message Mapping
+The content mode is chosen by the sender of the event, which is either the requesting or the responding party. Protocol interaction patterns that might allow solicitation of events using a particular content mode might be defined by an application, but are not defined here.
+
+The receiver of the event can distinguish between the two modes by inspecting the `CE-ContentType` property value. If the value exists and is prefixed with the CloudEvents media type `application/cloudevents`, indicating the use of a known [event format](#14-event-formats), the receiver uses *structured* mode, otherwise it defaults to *binary* mode.
+
+If a receiver detects the CloudEvents media type, but with an event format that it cannot handle, for instance `application/cloudevents+avro`, it MAY still treat the event as binary and forward it to another party as-is.
+
+The *structured* content mode keeps event metadata and data together, allowing simple forwarding of the same event across multiple routing hops, and across multiple transports.
+
+### 3.1 Binary Content Mode
+
+The *binary* content mode accommodates any shape of event data, and allows for efficient transfer and without transcoding effort.
+
+#### 3.1.1 Event Data Encoding
+
+The `data` attribute byte-sequence is used as the Apache Pulsar message payload.
+
+#### 3.1.2 Application Properties
+
+All [CloudEvents][CE] attributes with exception of `data` are individually mapped to and from the application properties section in Apache Pulsar messages.
+
+##### 3.1.2.1 Property Names
+
+The naming convention for the Pulsar application property mapping of attributes is:
+```
+* Each attribute name MUST be prefixed with "CE-"
+```
+
+Examples:
+```
+* `eventID` maps to `CE-EventID`
+* `cloudEventsVersion` maps to `CE-CloudEventsVersion`
+* `schemaURI` maps to `CE-SchemaURI`
+```
+
+For the `extensions` attribute, each entry of the `extensions` map is mapped to a separate application property. The `extensions` attribute itself is not mapped to an application property.
+
+The naming convention for the `extensions` application property mapping of attributes is:
+```
+* Each property name MUST be prefixed with "CE-X-"
+```
+
+Examples:
+```
+* `example` maps to `CE-X-Example`
+* `testExtension` maps to `CE-X-TestExtension`
+```
+Note: per Apache Pulsar protocol, property names are case-sensitive.
+
+##### 3.1.2.2 Property Values
+
+The value for each Pulsar property is constructed from the respective attribute’s [JSON value][RFC7159] representation, compliant with the [JSON event format][JSON-format] specification.
+
+#### 3.1.3 Examples
+
+This example shows the *binary* mode mapping of an event with an Apache Pulsar message:
+
+``` text
+--------------- message metadata ------------------
+
+properties: {
+    CE-CloudEventsVersion: "0.1"
+	  CE-EventType: "org.apache.pulsar.someevent"
```

Comment:
spacing on this line seems off

Reply:
do you need quotes around the names?

Reply:
yes. we need quotes around the names, since the names are strings. 

added quotes and fix the spacing.